编译不通过过

> Task :sample-plugin-app:transformClassesWithShadowTransformForRelease FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':sample-plugin-app:transformClassesWithShadowTransformForRelease'.
> java.util.zip.ZipException: duplicate entry: android/support/v7/app/ActionBar$DisplayOptions.class

--- a/gradle.properties
+++ b/gradle.properties
@@ -6,7 +6,7 @@
 # http://www.gradle.org/docs/current/userguide/build_environment.html
 # Specifies the JVM arguments used for the daemon process.
 # The setting is particularly useful for tweaking memory settings.
-org.gradle.jvmargs=-Xmx1536m
+org.gradle.jvmargs=-Xmx4096m
 # When configured, Gradle will run in incubating parallel mode.
 # This option should only be used with decoupled projects. More details, visit
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects

--- a/projects/sample/source/sample-plugin/sample-loader/build.gradle
+++ b/projects/sample/source/sample-plugin/sample-loader/build.gradle
@@ -46,6 +46,7 @@ afterEvaluate {
         buildTypes.findAll().each { buildType ->
             def buildTypeName = buildType.getName()
             def task = project.getTasksByName("check${buildTypeName.capitalize()}Classpath", false).first() as Task
+            task.dependsOn(":sample-host-lib:jarDebugPackage")
             task.dependsOn gradle.includedBuild('core').task(':common:jarDebugPackage')
             task.dependsOn gradle.includedBuild('dynamic').task(':dynamic-host:jarDebugPackage')
         }

--- a/projects/sdk/core/transform-kit/src/main/kotlin/com/tencent/shadow/core/transform_kit/AbstractTransform.kt
+++ b/projects/sdk/core/transform-kit/src/main/kotlin/com/tencent/shadow/core/transform_kit/AbstractTransform.kt
@@ -35,12 +35,14 @@ abstract class AbstractTransform(
 
     protected abstract val mTransformManager: AbstractTransformManager
     private val mOverrideCheck = OverrideCheck()
-    private val mDebugClassJar = File(project.buildDir, "transform-temp.jar")
+    private lateinit var mDebugClassJar: File
     private lateinit var mDebugClassJarZOS: ZipOutputStream
 
 
     private fun cleanDebugClassFileDir() {
-        mDebugClassJar.delete()
+        val transformTempDir = File(project.buildDir, "transform-temp")
+        transformTempDir.mkdirs()
+        mDebugClassJar = File.createTempFile("transform-temp", ".jar", transformTempDir)
         mDebugClassJarZOS = ZipOutputStream(FileOutputStream(mDebugClassJar))
     }
 

--- a/projects/sdk/core/transform-kit/src/main/kotlin/com/tencent/shadow/core/transform_kit/ClassTransform.kt
+++ b/projects/sdk/core/transform-kit/src/main/kotlin/com/tencent/shadow/core/transform_kit/ClassTransform.kt
@@ -161,6 +161,7 @@ abstract class ClassTransform(val project: Project) : Transform() {
      */
     open fun beforeTransform(invocation: TransformInvocation) {
         invocation.outputProvider.deleteAll()
+        inputSet.clear()
     }
 
     open fun afterTransform(invocation: TransformInvocation) {

--- a/projects/sdk/core/transform-kit/src/main/kotlin/com/tencent/shadow/core/transform_kit/OverrideCheck.kt
+++ b/projects/sdk/core/transform-kit/src/main/kotlin/com/tencent/shadow/core/transform_kit/OverrideCheck.kt
@@ -24,13 +24,17 @@ class OverrideCheck {
      */
     private val ctClassMap: MutableMap<String, CtClass> = hashMapOf()
 
+    private fun String.isKotlinClass(): Boolean {
+        return startsWith("kotlin")
+    }
+
     fun prepare(inputClasses: Set<CtClass>) {
         methodMap.clear()
         ctClassMap.clear()
 
         inputClasses.filter {
             //kotlinx里有一些方法的覆盖检查不出来，反正我们也不改它，就不检查了。
-            it.packageName.startsWith("kotlinx").not()
+            it.packageName.isKotlinClass().not()
         }.filter {
             try {
                 it.methods
@@ -65,7 +69,7 @@ class OverrideCheck {
 
         val errorResult: HashMap<String, MutableList<Method_OriginalDeclaringClass>> = hashMapOf()
         classNames.filter {
-            it.startsWith("kotlinx").not()
+            it.isKotlinClass().not()
         }.filter {
             val clazz = debugClassPool[it]!!
             try {

--- a/projects/sdk/core/transform-kit/src/main/kotlin/com/tencent/shadow/core/transform_kit/ReplaceClassName.kt
+++ b/projects/sdk/core/transform-kit/src/main/kotlin/com/tencent/shadow/core/transform_kit/ReplaceClassName.kt
@@ -32,6 +32,7 @@ object ReplaceClassName {
     private val errorResult: MutableMap<String, MutableMap<String, MutableSet<String>>> = mutableMapOf()
 
     fun resetErrorCount() {
+        mNewNames.clear()
         errorResult.clear()
     }
 

